tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_default_parameters.bal

```diff
@@ -300,6 +300,24 @@ function value2(record {|int a; int b;|} x) returns int {
     return x.a + x.b;
 }
 
+type F1 function() returns int;
+
+type F2 function() returns function() returns int;
+
+function baz(int a, F1 b = function() returns int { return a + 1; }) returns int {
+    return a + b();
+}
+
+function baz1(int a, F2 b = function() returns function() returns int { return function () returns int {return a;}; }) returns int {
+    F1 f = b();
+    return a + f();
+}
+
+function testParamAsDefaultForNextParameter() {
```

Comment:
Still doesn't reflect that it is part of an anon function.

Reply:
updated as `testParamUseAsValueInAnonFuncWithDefaultForNextParam`